src/net_processing.cpp

```diff
@@ -4420,10 +4420,10 @@ void PeerManagerImpl::MaybeSendPing(CNode& node_to, Peer& peer, std::chrono::mic
     }
 
     if (pingSend) {
-        uint64_t nonce = 0;
-        while (nonce == 0) {
-            GetRandBytes((unsigned char*)&nonce, sizeof(nonce));
-        }
+        uint64_t nonce;
+        do {
+            nonce = GetRand<uint64_t>();
+        } while (nonce == 0);
```

Comment:
d35019566a what is the reason for the `do/while` style change?

Reply:
It avoids a redundant assignment, and I think it makes more sense, although I would be happy to revert this change if you would strongly want that

Reply:
I wouldn't needless change it unless other reviewers prefer the change, but no strong opinion.

Reply:
I think the old code is slightly more elegant. There's no need to distinguish the first iteration here, so let's not.